GiniVision/Classes/Core/GiniScreenAPICoordinator+Analysis.swift

```diff
@@ -67,13 +67,18 @@ extension GiniScreenAPICoordinator: AnalysisDelegate {
         DispatchQueue.main.async { [weak self] in
             guard let `self` = self,
                 let message = message,
-                let action = action,
-                let firstDocument = self.documentRequests.first?.document else { return }
-            if self.analysisViewController == nil {
-                self.analysisViewController = self.createAnalysisScreen(withDocument: firstDocument)
-            }
+                let action = action else { return }
             
-            self.analysisViewController?.showError(with: message, action: action)
+            if let analysisViewController = self.analysisViewController {
+                analysisViewController.showError(with: message, action: { [weak self] in
```

Comment:
Is code is a duplicate to the one in the component API coordinator. In general a lot of code is duplicated between the two coordinators. I know that they are different ways to use the library and they have to do the same thing most of the time, but it would still be nice to combine what you can. I constantly see you fix something in the screen API and then remember to transfer it also to the component API. It's error prone.

Reply:
I know that is prone error, but they have to be different since they are different approaches. In this case I think it cannot be shared, but I always try to share the most code I can (since it is easier when implementing)